### PR TITLE
Migrate DetailsLink component to TS to improve code quality

### PR DESCRIPTION
**Pull request by Jinksi**
Part of #6929

#### Changes proposed in this Pull Request

This PR refactors the `DetailsLink` component and the shared function `getDetailsURL` to TypeScript to improve code quality and enhance developer experience.

#### Testing instructions

* Run `npx tsc --noEmit` and ensure no errors are returned.
* Ensure all tests pass with no changes to snapshot tests.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/components/details-link/index.tsx
Comment (Jinksi) [inline client/components/details-link/index.tsx:1]: The logic of this file is unchanged. Only TS types and comments have been added.

Comment (Jinksi) [reply client/components/details-link/index.tsx:1]: Sadly, the diff was lost – even using `git mv` which usually retains a nice diff.

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7086` or branch name `dev/details-link-ts-migration` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=dev/details-link-ts-migration) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 8f55ed8424dc5bc404c74cd4c2b094c70e7541a7
- Build time: 2023-08-31 22:01:15 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by shendy-a8c — APPROVED** [review]
Looks good.

- Confirmed by downloading `client/components/details-link/index.js` [from base branch](https://github.com/Automattic/woocommerce-payments/blob/1ba4330d04f488bcc63a8b691c044a23c783198f/client/components/details-link/index.js) and diff'd it with `client/components/details-link/index.tsx` to see that only comments and react import added.

- Ran `npx tsc --noEmit` and I didn't see error.

- Links to details page on dispute and deposits list page still work. `DetailsLink` is only used on deposit and dispute list page.

File: client/components/details-link/index.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Internal dependencies
+ */
+import { getAdminUrl } from 'wcpay/utils';
+
+/**
+ * The parent segment is the first part of the URL after the /payments/ path.
+ */
+type ParentSegment = 'deposits' | 'transactions' | 'disputes';
```
Comment (shendy-a8c) [inline client/components/details-link/index.tsx:18]: Just pointing out that `transactions` is only used in a unit test but no real use of `DetailsLink` that passes `transactions` to the `parentSegment` attribute.

**Pull request merged**